homeassistant/components/ambient_network/sensor.py

```diff
@@ -0,0 +1,299 @@
+    UnitOfPrecipitationDepth,
+    UnitOfPressure,
+    UnitOfSpeed,
+    UnitOfTemperature,
+    UnitOfVolumetricFlux,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import dt as dt_util
+
+from .const import CONF_MAC_ADDRESS, DOMAIN
+from .coordinator import AmbientNetworkDataUpdateCoordinator
+from .entity import AmbientNetworkEntity
+
+TYPE_AQI_PM25 = "aqi_pm25"
+TYPE_AQI_PM25_24H = "aqi_pm25_24h"
+TYPE_BAROMABSIN = "baromabsin"
+TYPE_BAROMRELIN = "baromrelin"
+TYPE_CO2 = "co2"
+TYPE_DAILYRAININ = "dailyrainin"
+TYPE_DEWPOINT = "dewPoint"
+TYPE_EVENTRAININ = "eventrainin"
+TYPE_FEELSLIKE = "feelsLike"
+TYPE_HOURLYRAININ = "hourlyrainin"
+TYPE_HUMIDITY = "humidity"
+TYPE_LASTRAIN = "lastRain"
+TYPE_LIGHTNING_DISTANCE = "lightning_distance"
+TYPE_LIGHTNING_PER_DAY = "lightning_day"
+TYPE_LIGHTNING_PER_HOUR = "lightning_hour"
+TYPE_MAXDAILYGUST = "maxdailygust"
+TYPE_MONTHLYRAININ = "monthlyrainin"
+TYPE_PM25 = "pm25"
+TYPE_PM25_24H = "pm25_24h"
+TYPE_SOLARRADIATION = "solarradiation"
+TYPE_TEMPF = "tempf"
+TYPE_UV = "uv"
+TYPE_WEEKLYRAININ = "weeklyrainin"
+TYPE_WINDDIR = "winddir"
+TYPE_WINDGUSTMPH = "windgustmph"
+TYPE_WINDSPEEDMPH = "windspeedmph"
+TYPE_YEARLYRAININ = "yearlyrainin"
+
+
+SENSOR_DESCRIPTIONS = (
+    SensorEntityDescription(
+        key=TYPE_AQI_PM25,
+        translation_key="pm25_aqi",
+        device_class=SensorDeviceClass.AQI,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=0,
+    ),
+    SensorEntityDescription(
+        key=TYPE_AQI_PM25_24H,
+        translation_key="pm25_aqi_24h_average",
+        device_class=SensorDeviceClass.AQI,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=0,
+    ),
+    SensorEntityDescription(
+        key=TYPE_BAROMABSIN,
+        translation_key="absolute_pressure",
+        native_unit_of_measurement=UnitOfPressure.INHG,
+        device_class=SensorDeviceClass.PRESSURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=2,
+    ),
+    SensorEntityDescription(
+        key=TYPE_BAROMRELIN,
+        translation_key="relative_pressure",
+        native_unit_of_measurement=UnitOfPressure.INHG,
+        device_class=SensorDeviceClass.PRESSURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=2,
+    ),
+    SensorEntityDescription(
+        key=TYPE_CO2,
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_MILLION,
+        device_class=SensorDeviceClass.CO2,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=2,
+    ),
+    SensorEntityDescription(
+        key=TYPE_DAILYRAININ,
+        translation_key="daily_rain",
+        native_unit_of_measurement=UnitOfPrecipitationDepth.INCHES,
+        device_class=SensorDeviceClass.PRECIPITATION,
+        state_class=SensorStateClass.TOTAL_INCREASING,
```

Comment:
Doesn't daily rain reset every day?

(Please leave the comments on device and state classes open so I have a quick way of seeing the discussion)

Reply:
(In general, do you want me to keep all comments open and you then resolve them?)

Reply:
Fixed.

Reply:
Oh, I by the way learnt something new, not sure if I told you now otherwise, if so, I am sorry. Total increasing can go back to 0. So if this value updates like: 1 -> 2 -> 50 -> 0. That's completely fine. That's seen as a new meter period.

But if the data doesn't fall back to zero (1 -> 2 -> 50 -> 1), it would be problematic.

Reply:
I don't think we can guarantee that it will fall back to zero. Even if it falls back to zero on the device, by the time we poll it next time, it may no longer be 0.

Reply:
Yea but that's fine. Like it should only go up, or completely down to 0

Reply:
Fixed.